velero-plugin-for-aws/object_store.go

```diff
@@ -350,17 +352,45 @@ func (o *ObjectStore) ListCommonPrefixes(bucket, prefix, delimiter string) ([]st
 		Delimiter: &delimiter,
 	}
 
+	tryListV1 := false
 	var ret []string
 	err := o.s3.ListObjectsV2Pages(req, func(page *s3.ListObjectsV2Output, lastPage bool) bool {
 		for _, prefix := range page.CommonPrefixes {
 			ret = append(ret, *prefix.Prefix)
 		}
+		if *page.IsTruncated && page.NextContinuationToken == nil {
```

Comment:
I'm not knowledgeable on the behavior of this API, but could you have a page that is truncated due to being the final page of results that wouldn't have a continuation token when everything is working as expected?

Or is a page only considered "truncated" when there are supposed to be more results on continued pages?

Reply:
@djwhatle truncated is true when there can be more results but the server couldn't send it on this page.